internal/pkg/vm/internal/runtime/runtime.go

```diff
@@ -8,27 +8,163 @@ import (
+	// Assert(bool)
+
+	// Review: we didnt have this before, why does the actor need to know the value supplied?
+	ValueSupplied() types.AttoFIL
+
+	// Review: if this is just for the `init` actor then it should not be here.
+	// Review: the init actor is a special system actor, its ok for it to have a different interface with a bit more power.
+    // CreateActor(
+    //     cid                actor.StateCID
+    //     a                  addr.Address
+    //     constructorParams  actor.MethodParams
+    // )
+
+	// Review: we provide this through the Storage()
+    // IpldGet(c ipld.CID) union {Bytes, error}
+	// IpldPut(x ipld.Object) ipld.CID
+	// Review: is the expectation fo storage to be global or actor specific? this has security implications down the line
+	Storage() Storage
+}
+
+// Review: more powerfull contract for the initactor.
+type innitInvocationContext interface {
+	InvocationContext
+	// Review: the spec has no error on this? what if the addr is already taken?
+	CreateActor(cid cid.Cid, addr address.Address, constructorParams MethodParams)
+}
+
+// Storage2 defines the storage module exposed to actors.
+// Review: check the old `Storage` for some more comments on why this looks like it does
+type Storage2 interface {
+	Put(interface{}) (cid.Cid, error)
+	Get(cid.Cid, interface{}) error
+}
+
+// Epoch is the current chain epoch.
+// Dragons: this is just to match names on the VM, Height got renamed to Epoch on the spec
+// TODO: rename types.BlockHeight to block.Epoch (issue: ???)
+type Epoch types.BlockHeight
+
+// Randomness is a string of random bytes
+// Review: placeholder for matching newtypes with the spec.
+type Randomness []byte
+
+// InvocInput are the params to invoke a method in an Actor.
+// Review: why is this a struct instead of arguments?
+type InvocInput struct {
+    To      address.Address
+    Method  types.MethodID
+    Params  MethodParams
+    Value   types.AttoFIL
+}
+
+// ExitCode is the exit code of a method executing inside the VM.
+// Review: do we have reserved codes?
+// TODO: this has type UVarint in the spec
+type ExitCode types.Int64
+
+// InvocOutput is waht os returned by an actor executing a method.
+// Review: multiple returns vs a return struct. There are some minor implications with swapping vm implementations written in another language.
+type InvocOutput struct {
+	ExitCode     ExitCode
+    ReturnValue  []byte
+}
+
+// MethodParam is the parameter to an Actor method already encoded for execution in the VM.
+type MethodParam []byte
+// MethodParams is a list of `MethodParam` to be passed into an Actor method.
+type MethodParams []MethodParam
+
 // Runtime defines the ABI interface exposed to actors.
 type Runtime interface {
+	// Review: this is not in the spec anymore
 	Message() *types.UnsignedMessage
+
+	// Review: the spec splits this up and brings the get/put to be first class
```

Comment:
I don't think this is fundamental, if it simplifies things to have a proxy object that should be fine too

Reply:
It's minor, it was more of an internal comment than anything. 
Personally I prefer reflecting the composition on the context, rather than flattening it out and turning it into a facade.